recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,226 @@
+            if self.settings.os != "Macos":
+                raise ConanInvalidConfiguration("{}/{} thread_system=pthread-apple is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
+        elif self.options.thread_system == "pthread":
+            if self.settings.os != "Linux":
+                raise ConanInvalidConfiguration("{}/{} thread_system=pthread is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
+        elif self.options.thread_system == "win32":
+            if self.settings.os != "Windows":
+                raise ConanInvalidConfiguration("{}/{} thread_system=win32 is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
+        elif self.options.thread_system == "pthread-emscripten":
+            if self.settings.os != "Emscripten":
+                raise ConanInvalidConfiguration("{}/{} thread_system=pthread-emscripten is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
 
     def _validate_boost_components(self):
         if self.settings.os != "Macos": return
         if self.settings.compiler != "apple-clang": return
         if Version(self.settings.compiler.version) >= "12": return
-        if self.options.boost_optional and self.options.boost_variant: return
+        if self.options.with_boost: return
         #
-        # On Apple we have to force the usage of boost.variant, because Apple's implementation of C++17 
+        # On Apple we have to force the usage of boost.variant and boost.optional, because Apple's implementation of C++17
         # is not complete.
         #
-        msg = "Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead. "
-        if not self.options.boost_optional and not self.options.boost_variant:
-            msg += "Try -o boost_optional=True -o boost_variant=True"
-        elif not self.options.boost_optional:
-            msg += "Try -o boost_optional=True."
-        else:
-            msg += "Try -o boost_variant=True."
-
-        raise ConanInvalidConfiguration(msg)
+        raise ConanInvalidConfiguration(f"Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead. Try -o {self.name}:with_boost=True.")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, '17')
-
-        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "9":
-            raise ConanInvalidConfiguration("Need GCC >= 9")
-
-        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "8":
-            raise ConanInvalidConfiguration("Need Clang >= 8")
+        # TODO: No cppstd check until this issue is solved
```

Comment:
This is resolved https://github.com/conan-io/conan/issues/12210#issuecomment-1320655950 :)